book/src/rfcs/benchmarking.md

```diff
@@ -0,0 +1,132 @@
+# Benchmarking RFC
+
+This document proposes a high level approach to benchmarking the crypto stack, mostly kimchi, snarky and pickles. The main concern is to set up an infrastructure where it is possible to integrate different kinds of benchmarks with minimal effort. While the specific benchmarks are not an immediate concern, a few benchmarks of interest will be mentioned later.
+
+## Types of benchmarks
+
+There are different metrics that we can collect, and different ways of presenting them depending of who is interested in them.
+
+### Audience
+
+We can differentiate to groups of people interested in benchmarks:
+
+- Developers working in the crypto stack, mostly our crypto team, where the main interest is to avoid regressions or make performance improvements.
+- People working outside of it, mostly users of our libraries, but could also include decision makers comparing other options or our marketing team, here the interest is more about more general metrics that can be compared with the competition or used to learn how to most efficiently use our technology.
```

Comment:
I don’t think this audience is entirely served by the current proposal. Decision makers for adopting this proof system stack at the application-level will want to understand the wall-clock time on commodity hardware of proving zkApps.

zkApp proofs (for now) are not proven in native execution even when they’re run server-side. Moreover, we probably need non-native benchmarks to help us internally decide how to prioritize building a zkapp flow that offloads some work to a native proving backend server-side.

Concretely, I think this translates to including in our benchmark matrix the following extra kinds of performance and memory tests:
{kimchi,pickles,snarky}-jsvm-{browser,server}

where “jsvm” is a mix of WASM and jsofocaml like SnarkyJS uses and browser means “run in the browser {V8 (Chrome), Safari, Firefox}” and server means running under a nodejs V8 environment. If we need to cut scope, just V8 for now in one of browser/server could be sufficient.

I am missing some context perhaps but I think for external audiences, these pickles web-based tests are either the most important benchmarks or second after native kimchi (for the purpose of being comparative against native proof systems).

Reply:
I added a bit at the end about leaving browser out of scope for now, but could be added at some point.